### PR TITLE
Single chan mode

### PR DESCRIPTION
**Pull request by LorenzMeier**
@DonLakeFlyer This implements single channel flight mode selection end-to-end. It needs more testing, but its absolutely there to build the UI support for it.

It will not affect already existing / configured systems and adds the new option.

### CODE REVIEW

**Comment by kd0aij** [comment]
In the case of a 6-position rotary switch, I would think the order of flight modes would be important, since you have to traverse them in order. So the UI might suggest reasonable sets of 6 to assign.
But one could also use OpenTX to select 9 modes using just 2 3-pos switches (3 sets of 3 modes) which might be nice, since we have at least 12 modes defined already. This would also have the desirable effect of removing the current constraints on mode switch priority. (e.g. RTL switch overrides everything)
In that case, the number of slots would need to be increased or made more flexible.

**Comment by LorenzMeier** [comment]
@DonLakeFlyer Any chance to get this UI wise in today? I really would like to get this out of the door if possible. I know you needed to do param infrastructure for it to work, but I hope that's addressed now.

**Comment by DonLakeFlyer** [comment]
Patience... I do need to sleep once in a while. I'm not like Gus :)

**Comment by DonLakeFlyer** [comment]
RC_MAP_FLTMODE is showing up in the "Default Group" in QGC. COM_FLTMODE# show up in Miscellaneous. Also RC_MAP_FAILSAFE is showing up in Miscellaneous

File: src/modules/commander/commander_params.c [hunk 1]
```diff
@@ -346,30 +346,27 @@ PARAM_DEFINE_INT32(COM_RC_IN_MODE, 0);
...
-PARAM_DEFINE_INT32(COM_FLTMODE_CH, 0);
-
 /**
  * First flightmode slot (1000-1160)
  *
  * If the main switch channel is in this range the
  * selected flight mode will be applied.
  *
- * @value 0 Unassigned
- * @value 1 MANUAL
```
Comment (DonLakeFlyer) [inline src/modules/commander/commander_params.c]: Do the Flight Mode names need to be in all caps? They are screaming at me!

**Comment by DonLakeFlyer** [comment]
The naming of the flight modes should match the naming used by QGC: https://github.com/mavlink/qgroundcontrol/blob/master/src/FirmwarePlugin/PX4/PX4FirmwarePlugin.cc#L74

**Comment by DonLakeFlyer** [comment]
Can you push the param versioning pull before this one. Then rebase, and bump param minor version so a flash from this will trigger override of QGC internal meta data.

**Comment by LorenzMeier** [comment]
@DonLakeFlyer Rebased and pushed.

**Comment by DonLakeFlyer** [comment]
I tried flying with this to test new QGC flight mode stuff:
- Kept telling me "sensors need inspection" even though as far as QGC parameter testing is concerned everything is fine. Tried to redo do a compass cal. No help. Not a very helpful error message from firmware.
- Complained about RC_MAP_MODE_SW not being set, rc cal check failed

**Comment by DonLakeFlyer** [comment]
Switched to Stable. Still getting sensors need inspection. Not sure how to get my PixRacer 250 back in the air again!

**Comment by DonLakeFlyer** [comment]
FYI: Flashed to prebuilt master and everything is happy again, strange.

**Comment by LorenzMeier** [comment]
@DonLakeFlyer Fixed state pushed to master.

**Comment by kd0aij** [comment]
This kind of unexplained behavior is a major issue affecting "user friendliness". If it even confuses the developers, normal users won't have any chance of success.

**Comment by LorenzMeier** [comment]
@kd0aij Sorry, I don't know what you mean.

**Comment by kd0aij** [comment]
oops, replied on a stale window... I was looking at:
"Switched to Stable. Still getting sensors need inspection. Not sure how to get my PixRacer 250 back in the air again!
followed by:
"FYI: Flashed to prebuilt master and everything is happy again, strange."

**Pull request closed without merging**